File: generated_code/optimized/runtime/java/3.java
import java.util.*;

class Solution {
    public boolean belowZero(List<Integer> operations) {
        int balance = 0;
        if (operations instanceof RandomAccess) {
            for (int i = 0, n = operations.size(); i < n; i++) {
                balance += operations.get(i);
                if (balance < 0) return true;
            }
        } else {
            for (Integer op : operations) {
                balance += op;
                if (balance < 0) return true;
            }
        }
        return false;
    }
}

File: generated_code/optimized/balanced/detailed/cpp/3.cpp
#include <vector>

bool below_zero(std::vector<int> operations) {
    int sum = 0;
    const int* p = operations.data();
    const int* end = p + operations.size();

    while (p != end) {
        sum += *p++;
        if (sum < 0) return true;
    }

    return false;
}
